packages/care-package/import-vapp/src/BuildActions.ts

```diff
@@ -0,0 +1,116 @@
+import * as careDef from '@vcd/care-package-def';
+import { JSONSchema7 } from 'json-schema';
+import {LegacyApiClient} from '@vcd/node-client/lib/legacy.api.client';
+import {Configuration} from './Configuration';
+import {
+    AdminCatalogType, CatalogItemType, InstantiateVAppTemplateParamsType,
+    QueryResultRecordsType,
+    UploadVAppTemplateParamsType
+} from '@vcd/bindings/vcloud/api/rest/schema_v1_5';
+import {CloudDirectorConfig} from '@vcd/node-client';
+
+export class BuildActions implements careDef.BuildActions {
+
+    legacyApiClient: LegacyApiClient;
+
+    getInputSchema(action: string): JSONSchema7 {
+        if (action === 'generate') {
+            return {
+                type: 'object',
+                properties: {
+                    name: {
+                        type: 'string',
+                        description: 'element name'
+                    }
+                }
+            };
+        }
+        return null;
+    }
+
+    getConfiguration?(): any {
+        return {
+            catalog: 'care-package',
+            vAppTemplateName: '',
+            vAppTemplateURL: ''
+        };
+    }
+
+    private async executeRequests(element: careDef.ElementSource) {
+        console.log('executeRequests');
+
+        const configuration: Configuration = element.configuration;
+        console.log('configuration: ' +  JSON.stringify(configuration));
+
+        const catalogId = await this.getOrCreateCatalogId(configuration);
+
+        await this.uploadVAppTemplate(catalogId, configuration);
+    }
+
+    private async getOrCreateCatalogId(configuration: Configuration) {
+        const catalogQuery = '/api/query?type=catalog&filter=name==' + configuration.catalog;
```

Comment:
Is that expected "name==" instead of "name="?

Reply:
Yes, the reason:
filter = (name==value)

the single = is used for the query params the == for inner ones